awx/ui_next/src/screens/Project/Project.jsx

```diff
@@ -0,0 +1,272 @@
+      prevProps.location.pathname.startsWith(url) &&
+      prevProps.location !== location &&
+      location.pathname === `${url}details`
+    ) {
+      await this.loadProject();
+    }
+  }
+
+  async loadProjectAndRoles() {
+    const { match, setBreadcrumb } = this.props;
+    const id = parseInt(match.params.id, 10);
+
+    this.setState({ contentError: null, hasContentLoading: true });
+    try {
+      const [{ data }, notifAdminRes] = await Promise.all([
+        ProjectsAPI.readDetail(id),
+        OrganizationsAPI.read({
+          page_size: 1,
+          role_level: 'notification_admin_role',
+        }),
+      ]);
+      const [auditorRes, adminRes] = await Promise.all([
+        OrganizationsAPI.read({
+          id: data.organization,
+          role_level: 'auditor_role',
+        }),
+        OrganizationsAPI.read({
+          id: data.organization,
+          role_level: 'admin_role',
+        }),
+      ]);
+      setBreadcrumb(data);
+      this.setState({
+        project: data,
+        isNotifAdmin: notifAdminRes.data.results.length > 0,
+        isAuditorOfThisOrg: auditorRes.data.results.length > 0,
+        isAdminOfThisOrg: adminRes.data.results.length > 0,
+      });
+    } catch (err) {
+      this.setState({ contentError: err });
+    } finally {
+      this.setState({ hasContentLoading: false });
+    }
+  }
+
+  async loadProject() {
+    const { match, setBreadcrumb } = this.props;
+    const id = parseInt(match.params.id, 10);
+
+    this.setState({ contentError: null, hasContentLoading: true });
+    try {
+      const { data } = await ProjectsAPI.readDetail(id);
+      setBreadcrumb(data);
+      this.setState({ project: data });
+    } catch (err) {
+      this.setState({ contentError: err });
+    } finally {
+      this.setState({ hasContentLoading: false });
+    }
+  }
+
+  render() {
+    const { location, match, me, history, i18n } = this.props;
+
+    const {
+      project,
+      contentError,
+      hasContentLoading,
+      isInitialized,
+      isNotifAdmin,
+      isAuditorOfThisOrg,
+      isAdminOfThisOrg,
+    } = this.state;
+
+    const canSeeNotificationsTab =
+      me.is_system_auditor || isNotifAdmin || isAuditorOfThisOrg;
+    const canToggleNotifications =
+      isNotifAdmin &&
+      (me.is_system_auditor || isAuditorOfThisOrg || isAdminOfThisOrg);
+
+    const tabsArray = [
+      { name: i18n._(t`Details`), link: `${match.url}/details`, id: 0 },
+      { name: i18n._(t`Access`), link: `${match.url}/access`, id: 1 },
+    ];
+
+    if (canSeeNotificationsTab) {
+      tabsArray.push({
+        name: i18n._(t`Notifications`),
+        link: `${match.url}/notifications`,
+        id: 2,
+      });
+    }
+
+    tabsArray.push(
+      {
+        name: i18n._(t`Job Templates`),
+        link: '/job_templates',
+        id: canSeeNotificationsTab ? 3 : 2,
+      },
+      {
+        name: i18n._(t`Schedules`),
+        link: '/schedules',
+        id: canSeeNotificationsTab ? 4 : 3,
```

Comment:
could maybe just assign the ids to the tab objects at the end: 

```js
tabsArray.forEach((tab, n) => {
  tab.id = n;
}); 
```

This would ensure the tabs have the right ids even if we shift them around / add more in the future.

Reply:
Yea, I like that better.  I'll apply it in all the places.